dll/win32/shell32/precomp.h

```diff
@@ -139,8 +139,11 @@ AddPropSheetPageCallback(HPROPSHEETPAGE hPage, LPARAM lParam)
     return FALSE;
 }
 
-HRESULT WINAPI
-Shell_DefaultContextMenuCallBack(IShellFolder *psf, IDataObject *pdtobj);
+HRESULT
+SHELL32_ShowPropertiesDialog(IDataObject *pdtobj);
+HRESULT
+SHELL32_DefaultContextMenuCallBack(IShellFolder *psf, IDataObject *pdo, UINT msg);
+#define SHELL_ExecuteControlPanelCPL(hwnd, cpl) SHRunControlPanel((cpl), (hwnd))
```

Comment:
The first 2 are `SHELL32`, the last one is `SHELL`.
Is there any particular reason for this?

Reply:
I don't think there is official guidance for this but I will tell you my interpretation of these names.

- SH and Shell_, these are Microsoft functions.
- SHELL32_ is a function that can only be called from other code inside shell32.
- SHELL_ is a more general shell function that can be moved to shellutils.h if other modules need it.